common/management/commands/create_test_data.py

```diff
@@ -0,0 +1,1775 @@
+        if applicant_type == APPLICANT_TYPES.APPLICANT_TYPE_INDIVIDUAL
+        else (applicant_type, next(organisations_iter))
+        for applicant_type in applicant_types
+    ]
+
+
+@with_logs(
+    text_entering="Creating application events...",
+    text_exiting="Application events created!",
+)
+def _create_application_events(
+    applications: list[Application],
+    age_groups: list[AgeGroup],
+    reservation_purposes: list[ReservationPurpose],
+    reservation_units: list[ReservationUnit],
+) -> list[ApplicationEvent]:
+    application_events: list[ApplicationEvent] = []
+
+    for application in applications:
+        for _ in range(random.randint(1, 3)):
+            name = faker_fi.word()
+            min_duration = random.randint(1, 2)
+            max_duration = random.randint(min_duration, 5)
+
+            event = ApplicationEvent(
+                name=name,
+                name_fi=name,
+                name_en=faker_en.word(),
+                name_sv=faker_sv.word(),
+                num_persons=random.randint(1, 100),
+                age_group=random.choice(age_groups),
+                min_duration=timedelta(hours=min_duration),
+                max_duration=timedelta(hours=max_duration),
+                events_per_week=random.randint(1, 10),
+                biweekly=random.choices([True, False], weights=[1, 5])[0],
+                begin=application.application_round.reservation_period_begin,
+                end=application.application_round.reservation_period_end,
+                application=application,
+                purpose=random.choice(reservation_purposes),
+            )
+            application_events.append(event)
+
+    application_events = ApplicationEvent.objects.bulk_create(application_events)
+    _create_event_reservation_units(application_events, reservation_units)
+    _create_application_event_schedules(application_events)
+    return application_events
+
+
+@with_logs(
+    text_entering="Creating application reservation units...",
+    text_exiting="Application reservation units created!",
+)
+def _create_event_reservation_units(
+    application_events: list[ApplicationEvent],
+    reservation_units: list[ReservationUnit],
+) -> list[EventReservationUnit]:
+    event_units: list[EventReservationUnit] = []
+    for application_event in application_events:
+        amount = random.randint(1, 4)
+        units: list[ReservationUnit] = random.sample(reservation_units, k=amount)
+
+        for i, unit in enumerate(units):
+            event_unit = EventReservationUnit(
+                priority=i,
+                application_event=application_event,
+                reservation_unit=unit,
+            )
+            event_units.append(event_unit)
+
+    return EventReservationUnit.objects.bulk_create(event_units)
+
+
+@with_logs(
+    text_entering="Creating application event schedules...",
+    text_exiting="Application event schedules created!",
+)
+def _create_application_event_schedules(
```

Comment:
Allocated schedules only for `APPROVED` application events; `CREATED` and `DECLINED` have none.

Reply:
Done.